snapcraft/plugins/catkin.py

```diff
@@ -277,6 +283,22 @@ def build(self):
         self._finish_build()
 
     def _prepare_build(self):
+        # Fix all shebangs to use the in-snap python.
+        common.replace_in_file(self.rosdir, re.compile(r''),
+                               re.compile(r'#!.*python'),
+                               r'#!/usr/bin/env python')
+
+        # Also replace the python usage in 10.ros.sh to use the in-snap python.
+        ros10_file = os.path.join(self.rosdir,
+                                  'etc/catkin/profile.d/10.ros.sh')
+        if os.path.isfile(ros10_file):
```

Comment:
Ah, this was tricky. I missed the 'r+' on the mode-- we open for read/write. We need to know if it exists before we try, or we'll create it (and we don't want to if it's not already there). I've waffled back on this.